pyccel/codegen/printing/ccode.py

```diff
@@ -1106,8 +1106,10 @@ def _print_IndexedElement(self, expr):
                     if isinstance(ind, Slice):
                         inds[i] = self._new_slice_with_processed_arguments(ind, PyccelArraySize(base, i),
                             allow_negative_indexes)
+                        inds[i].slice_type = Slice.Range
```

Comment:
Shouldn't the `_new_slice_with_processed_arguments` function set the `slice_type` to avoid duplication?

Reply:
https://github.com/pyccel/pyccel/pull/1299/commits/ee8cf11658638a9af0b93edc9f4abc71323b9f89